core/src/fields.rs

```diff
@@ -570,34 +574,45 @@ impl MongoFields {
 impl MongoStatement for MongoFields {
     // Move the cursor to the next document and update the current row.
     // Return true if moving was successful, false otherwise.
-    fn next(&mut self, mongo_connection: Option<&MongoConnection>) -> Result<bool> {
+    fn next(
+        &mut self,
+        mongo_connection: Option<&MongoConnection>,
+    ) -> Result<(bool, Option<Vec<Error>>)> {
         match self.field_name_filter.as_ref() {
             None => {
                 self.current_field_for_collection += 1;
-                Ok(
-                    (self.current_field_for_collection as usize) < self.current_col_metadata.len()
-                        || self.get_next_metadata(mongo_connection.unwrap()),
-                )
+                match (self.current_field_for_collection as usize) < self.current_col_metadata.len()
+                {
+                    true => Ok((true, None)),
+                    false => self.get_next_metadata(mongo_connection.unwrap()),
+                }
             }
             Some(filter) => {
                 let filter = filter.clone();
                 loop {
                     self.current_field_for_collection += 1;
-                    if (self.current_field_for_collection as usize
-                        >= self.current_col_metadata.len())
-                        && !self.get_next_metadata(mongo_connection.unwrap())
+                    let mut e: Vec<Error> = vec![];
+                    if self.current_field_for_collection as usize >= self.current_col_metadata.len()
                     {
-                        return Ok(false);
+                        let next = self.get_next_metadata(mongo_connection.unwrap());
+                        if let Ok((b, errors)) = next {
```

Comment:
Can you rename the variables and use something like `Ok((b, warnings))` or maybe even `Ok((has_more_coll, warnings))`?